tests/test_primitives/test_point/test_point_serializers.py

```diff
@@ -118,3 +118,31 @@ def test_point_not_on_curve():
     from cryptography.exceptions import InternalError
     with pytest.raises(InternalError):
         Point.from_bytes(point_on_koblitz256_but_not_P256.to_bytes(), curve=SECP256R1)
+
+
+def test_serialize_point_at_infinity():
+
+    p = Point.gen_rand()
+    point_at_infinity = p - p
+    
+    with pytest.raises(InternalError) as e:
+        _bytes_point_at_infinity = point_at_infinity.to_bytes()
+
+    # We want to catch specific InternalExceptions:
+    # - Point at infinity (code 107)
+    assert e.value.err_code[0].reason == 106
```

Comment:
Do you think we should start catching these errors and build our own error system to raise specific errors from OpenSSL except cleaner?

Probably works better with building an arithmetic library.

Reply:
Yep, good idea. Similar code is in other functions (e.g., `Point.unsafe_hash_to_point`), and it's ugly as hell. I just created #197 with that.